server/models/user.py

```diff
@@ -1,7 +1,55 @@
 from DB.database import Database
+import bcrypt
 
 class User:
-	def __init__(self) -> None:
-		self.db = Database()
-		self.collection_name = "User"
-		
\ No newline at end of file
+	def __init__(self):
+		self.db = Database().get_database()
+
+	def login(self,email,password):
+		try:
+			# bytes = password.encode("utf-8")
+			# salt = "b'$2b$12$lyTSirFB5Qeqv0VGl9GmNe'"
+			# hash = str(bcrypt.hashpw(bytes, salt))
+			# print(salt)
+			# print("b'$2b$12$VV7W1vaScWnMGMtaN1nd2O14qQu/TwvyUNRW.4MprjJmKXJmRaYTi'")
+			user = self.db.User.find_one({"emailid":email})
+			if(user["password"]==password):
+				user.pop("password")
+				return user
+			return {}
+		except:
+			return {}
+
+	def register(self, user_data):
+		try:
+			table = self.db.User
+			password = user_data["password"]
+			# bytes = password.encode("utf-8")
+			# salt = "b'$2b$12$lyTSirFB5Qeqv0VGl9GmNe'"
+			# hash = str(bcrypt.hashpw(bytes, salt))
+			# print(hash)
+			table.insert_one({
+				"emailid": user_data["emailid"],
+				"password": password,
+				"gender": user_data["gender"],
+				"is_admin": False,
+			})
+			return self.fetch_user(user_data["emailid"])
+		except Exception as e:
+			return {}
```

Comment:
Error handling needs to be improved. Print the exception details after catching the exception so you have an idea what went wrong